runtime/src/bank_forks.rs

```diff
@@ -86,8 +86,55 @@ impl Index<u64> for BankForks {
 
 impl BankForks {
     pub fn new_rw_arc(bank: Bank) -> Arc<RwLock<Self>> {
+        let bank = Arc::new(bank);
```

Comment:
quick question on interface here, which might have simplified the change set. Seems this function could take either a `Bank` or an `Arc<Bank>`. Could do something like this:

```suggestion
    pub fn new_rw_arc(bank: impl Into<Arc<Bank>>) -> Arc<RwLock<Self>> {
        let bank = bank.into();
```

however, given you've already done the changes, I'm fine just not supporting it too.

Reply:
Yeah, I waffled a bit on this after we swapped DM's, but I ended up sticking with what I had. My reasoning:
- `BankForks` is essentially mutating banks at the end of `BankForks::new_rw_arc()` (the lines that touch the load_program_cache)
- While this code is outside of `Bank`, it appears to essentially be initialization for `Bank`. But, it is separate due to circular dependency (we need a `Bank` to create `BankForks`, but we need a `BankForks` in order to finish setting up the `loaded_program_cache` in a `Bank`)
- Assuming you agree with my sentiment about "initialization code", I think `BankForks` should ONLY take a `Bank`. Taking an `Arc<Bank>` could imply some other thread has a reference to that `Bank` and could be operating on it before we finish the initialization.

So, the interface of requiring a `Bank` is saying let `BankForks` do what it needs to do while we know that this is the only thread accessing the `Bank`, and then allow fetching the `Arc<Bank>` out by whoever wants it after `BankForks` has been created

Reply:
That sounds really reasonable!